### PR TITLE
Tidy up aks installer

### PR DESCRIPTION
**Pull request by pierskarsenbarg**

### CODE REVIEW

File: aks-hosted/01-infrastructure/database.ts [hunk 1]
```diff
@@ -27,27 +26,26 @@ export class Database extends pulumi.ComponentResource {
...
+                publicNetworkAccess: "Enabled", // allow traffic from vnet (not public) based on firewall rule below;
                 sslEnforcement: "Disabled",
                 storageProfile: {
                     backupRetentionDays: 7,
                     geoRedundantBackup: "Disabled",
                     storageAutogrow: "Enabled",
                     storageMB: 51200,
                 },
-                version: "5.7",
+                version: "8.0",
```
Comment (joeduffy) [inline aks-hosted/01-infrastructure/database.ts:48]: Just for my own education, we are actively testing against 8.0 these days, right? For some reason, I thought we were still on 5.7.

**Comment by joeduffy** [comment]
I want to ensure I am following along. 90%+ of these changes are formatting (spaces) and non-semantic changes (importing names differently), as far as I can tell. The meaningful changes I found were:

* MySQL version [changed from 5.7 to 8.0](https://github.com/pulumi/pulumi-self-hosted-installers/pull/97/files#diff-ae45425cdd8be50ccf7f8accbf006bb377425576d7440a734789632eea7deb61R48)
* [The VNet routing change](https://github.com/pulumi/pulumi-self-hosted-installers/pull/97/files#diff-ae45425cdd8be50ccf7f8accbf006bb377425576d7440a734789632eea7deb61R72-R75) which presumably was the culprit for connectivity issues?
* Kubernetes version [changed from 1.21.9 to 1.26.3](https://github.com/pulumi/pulumi-self-hosted-installers/pull/97/files#diff-c073c9c4de0967ce61f29c57f0b3e044d90dba9e9a1671591a1fc4889b3fb7d8R65)
* The NGINX ingress chart version [changed from 3.31.0 to 4.6.1](https://github.com/pulumi/pulumi-self-hosted-installers/pull/97/files#diff-8b8191ca60afc02199b3bd8ec86330097ebf796f60dc775494a4e3337eebb886R26), a few tags changed from `"beta.kubernetes.io/os"` to `"kubernetes.io/os"`, and the `"externalTrafficPolicy"` setting was set to `"Local"`

Was that everything noteworthy? Just hoping that helps other folks focus on reviewing the meaningful changes.

File: aks-hosted/01-infrastructure/database.ts (hunk 1 above)
Comment (phillipedwards) [reply aks-hosted/01-infrastructure/database.ts:48]: I believe we are testing against both 5.7 and 8.0 although 5.7 is EOL in October.

**Comment by phillipedwards** [comment]
@joeduffy like you said, the large majority of the changes are formatting or non-semantic changes to reduce import sizes. I don't believe there are any vnet changes of consequence.

The only changes of consequence are:
- upgrade from 5.7 to 8.0. 5.7 is EOL 10/23 and 8.0 is fully supported.
- K8s upgraded to 1.26.3. Azure doesn't support anything less than 1.22 in most regions.
- nginx helm chart upgraded to latest (4.6.1), beta removed, and externalTrafficPolicy is set to local due to an Azure peculiarity.

While we don't currently have any integration tests, the app was tested manually.

**Review by phillipedwards — APPROVED** [review]
LGTM

**Pull request merged**